nistats/second_level_model.py

```diff
@@ -331,39 +275,56 @@ def fit(self, second_level_input, first_level_conditions=None,
 
         return self
 
-    def compute_contrast(self, contrast_def, stat_type=None,
-                         output_type='z_score'):
+    def compute_contrast(self, contrast_def='contrast',
+                         first_level_contrast=None,
+                         stat_type=None, output_type='z_score'):
         """Generate different outputs corresponding to
         the contrasts provided e.g. z_map, t_map, effects and variance.
 
         Parameters
         ----------
-        contrast_def : str or array of shape (n_col)
-            where ``n_col`` is the number of columns of the design matrix,
+        contrast_def: str or array of shape (n_col), optional
+            Where ``n_col`` is the number of columns of the design matrix,
             The string can be a formula compatible with the linear constraint
             of the Patsy library. Basically one can use the name of the
             conditions as they appear in the design matrix of
             the fitted model combined with operators /*+- and numbers.
             Please checks the patsy documentation for formula examples:
             http://patsy.readthedocs.io/en/latest/API-reference.html#patsy.DesignInfo.linear_constraint
+            By default it returns the group contrast of the main effect.
 
-        stat_type : {'t', 'F'}, optional
-            type of the contrast
+        first_level_contrast: str or array of shape (n_col) with respect to
+                              FirstLevelModel, optional
+            In case a list of FirstLevelModel was provided as
+            second_level_input, we have to provide a contrast to apply to
+            the first level models to get the corresponding list of images
+            desired, that would be tested at the second level.
 
-        output_type : str, optional
+        stat_type: {'t', 'F'}, optional
+            Type of the contrast
```

Comment:
the secon-d-level contrast (note that it has to be very explicit whether we're talking about first or second level contrast).